### PR TITLE
Add cuda samples

### PR DESCRIPTION
**Pull request by huebner-m**
Add CUDAsamples for CUDA version 11.3 (see https://github.com/NVIDIA/cuda-samples).
Some samples are filtered out due to extensive dependencies. Sanity checks are limited to the existence of paths, commands may fail due to missing compatibility libraries that might be needed to use CUDA 11.3 in combination with the available driver version.

### CODE REVIEW

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen2

File: easybuild/easyconfigs/c/CUDAsamples/CUDAsamples-11.3.eb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+dependencies = [
+    ('CUDA', '11.3.1', '', True),
+]
+
+# Get rid of pre-built Windows DLLs and only build deviceQuery for now.
+prebuildopts = "rm -r bin/win64 && "
+
+# Filter out samples that require extensive dependencies.
+buildopts = "HOST_COMPILER=g++ FILTER_OUT='Samples/EGLStream_CUDA_Interop/Makefile Samples/boxFilterNPP/Makefile Samples/cannyEdgeDetectorNPP/Makefile Samples/cudaNvSci/Makefile Samples/cudaNvSciNvMedia/Makefile Samples/simpleGL/Makefile Samples/warpAggregatedAtomicsCG/Makefile Samples/simpleVulkan/Makefile Samples/simpleVulkanMMAP/Makefile Samples/streamOrderedAllocationIPC/Makefile Samples/vulkanImageCUDA/Makefile'"
```
Comment (ocaisa) [inline easybuild/easyconfigs/c/CUDAsamples/CUDAsamples-11.3.eb:24]:
```suggestion
local_filters = "Samples/EGLStream_CUDA_Interop/Makefile "
local_filters += "Samples/boxFilterNPP/Makefile "
local_filters += "Samples/cannyEdgeDetectorNPP/Makefile "
local_filters += "Samples/cudaNvSci/Makefile "
local_filters += "Samples/cudaNvSciNvMedia/Makefile "
local_filters += "Samples/simpleGL/Makefile "
local_filters += "Samples/warpAggregatedAtomicsCG/Makefile "
local_filters += "Samples/simpleVulkan/Makefile "
local_filters += "Samples/simpleVulkanMMAP/Makefile "
local_filters += "Samples/streamOrderedAllocationIPC/Makefile "
local_filters += "Samples/vulkanImageCUDA/Makefile"
buildopts = "HOST_COMPILER=g++ FILTER_OUT='%s'" % local_filters 
```

File: easybuild/easyconfigs/c/CUDAsamples/CUDAsamples-11.3.eb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+local_filters += "Samples/streamOrderedAllocationIPC/Makefile "
+local_filters += "Samples/vulkanImageCUDA/Makefile"
+buildopts = "HOST_COMPILER=g++ FILTER_OUT='%s'" % local_filters 
+
+files_to_copy = [
+    (['bin/%s/linux/release/*' % ARCH], 'bin'),
+    'LICENSE',
+]
+
+local_binaries = ['deviceQuery']
```
Comment (ocaisa) [inline easybuild/easyconfigs/c/CUDAsamples/CUDAsamples-11.3.eb:42]:
```suggestion
local_binaries = ['deviceQuery', 'matrixMul', 'bandwidthTest', 'cudaOpenMP']
```

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen2

File: easybuild/easyconfigs/c/CUDAsamples/CUDAsamples-11.3.eb [hunk 3]
```diff
@@ -0,0 +1,53 @@
+easyblock = 'MakeCp'
+
+name = 'CUDAsamples'
```
Comment (ocaisa) [inline easybuild/easyconfigs/c/CUDAsamples/CUDAsamples-11.3.eb:3]: We should retain the name as close as possible to what they use themselves ("CUDA Samples") so either `CUDA-Samples` or `CUDA_Samples`

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/c/CUDAsamples/CUDAsamples-11.3.eb (hunk 3 above)
Comment (huebner-m) [reply easybuild/easyconfigs/c/CUDAsamples/CUDAsamples-11.3.eb:3]: This name should be reflected in the naming of the folder and the file, right?

Comment (ocaisa) [reply easybuild/easyconfigs/c/CUDAsamples/CUDAsamples-11.3.eb:3]: I think you can change the name locally and use `--update-pr`

Comment (ocaisa) [reply easybuild/easyconfigs/c/CUDAsamples/CUDAsamples-11.3.eb:3]: https://docs.easybuild.io/en/latest/Integration_with_GitHub.html#updating-existing-pull-requests-update-pr and I know there is a way to delete the old easyconfig too but I can't remember

Comment (ocaisa) [reply easybuild/easyconfigs/c/CUDAsamples/CUDAsamples-11.3.eb]: Ah, the filename is wrong anyway, it should be `CUDA-Samples/CUDA-Samples-GCC-10.3.0-CUDA-11.3.1.eb`

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen2

**Review by ocaisa — DISMISSED** [review]
LGTM

**Comment by ocaisa** [comment]
CI seems confused, closing and reopening

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen2

**Review by ocaisa — APPROVED** [review]

**Comment by ocaisa** [comment]
Going in, thanks @huebner-m !

**Pull request merged**